integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/vector/LuceneVectorFieldIT.java

```diff
@@ -0,0 +1,464 @@
+						.select( f -> f.id() )
+						.where( f -> f.knn( 1 ).field( "byteVector" )
+								.matching( bytes( 4, (byte) 5 ) ) )
+		).hasTotalHitCount( 1 );
+
+		assertThatQuery(
+				index.createScope().query()
+						.select( f -> f.id() )
+						.where( f -> f.knn( 1 ).field( "floatVector" )
+								.matching( floats( 8, 0.051f ) ) )
+		).hasTotalHitCount( 1 );
+	}
+
+	@Test
+	void simpleVectorPredicateWithFilter() {
+		// took the sample data and query from this example https://opensearch.org/docs/latest/search-plugins/knn/filter-search-knn/#using-a-lucene-k-nn-filter
+		// to see if we'll get the same results... and looks like we do :smile:
+		SimpleMappedIndex<PredicateIndexBinding> index = SimpleMappedIndex.of( PredicateIndexBinding::new )
+				.name( "predicateIndex" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimplePredicate( index );
+
+
+		int k = 3;
+		SearchQuery<float[]> query = index.createScope().query()
+				.select(
+						f -> f.field( "location", float[].class )
+				)
+				.where( f -> f.knn( k )
+						.field( "location" )
+						.matching( 5f, 4f )
+						.filter( f.range().field( "rating" ).between( 8, 10 ) )
+						.filter( f.terms().field( "parking" ).matchingAny( true ) )
+				).toQuery();
+
+		List<float[]> result = query.fetchAll().hits();
+
+		assertThat( result )
+				.hasSize( k ) // since that is how many neighbors we were asking for in the predicate
+				.containsExactly(
+						new float[] { 4.9f, 3.4f },
+						new float[] { 6.4f, 3.4f },
+						new float[] { 3.3f, 4.5f }
+				);
+	}
+
+	@Test
+	void insideOtherPredicate() {
+		SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new ).name( "index" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimple( index );
+
+		SearchQuery<String> query = index.createScope().query()
+				.select(
+						f -> f.field( "string", String.class )
+				)
+				.where( f -> f.bool()
+						.must( f.knn( 5 ).field( "byteVector" ).matching( bytes( 4, (byte) 5 ) ) )
+						.must( f.match().field( "string" ).matching( "keyword1" ) )
+				)
+				.toQuery();
+
+		List<String> result = query.fetchAll().hits();
+		assertThat( result )
+				.hasSize( 1 )
+				.containsOnly( "keyword1" );
+	}
+
+	@Test
+	void nestedVector() {
+		SimpleMappedIndex<NestedIndexBinding> index = SimpleMappedIndex.of( NestedIndexBinding::new ).name( "index" );
+		setupHelper.start().withIndexes( index ).setup();
+		initNestedIndex( index );
+
+		assertThat(
+				index.createScope().query()
+						.select( f -> f.composite()
+								.from(
+										f.id(),
+										f.object( "nested" )
+												.from(
+														f.field( "nested.byteVector" ),
+														f.field( "nested.floatVector" )
+												)
+												.asList()
+												.multi()
+								).asList()
+						).where(
+								f -> f.nested( "nested" )
+										.add( f.knn( 1 ).field( "nested.byteVector" ).matching( bytes( 2, (byte) -120 ) ) )
```

Comment:
I'm surprised this even works, considering it's not supported in Elasticsearch... does this also work with a multi-valued nested field? Because that's kind of the point of nested fields; single-valued nested fields are not useful.

If it does work, maybe it performs badly?

Reply:
Yeah, honestly I didn't know what to expect there, and it was more of "let's try and see if it will not fail" 😄